src/databaseInterface/src/main/kotlin/org/vaccineimpact/api/db/PostgresCopyWriter.kt

```diff
@@ -0,0 +1,50 @@
+package org.vaccineimpact.api.db
+
+import java.io.Closeable
+import java.io.Flushable
+import java.io.OutputStream
+import java.io.Writer
+
+/**
+ * At present this class only writes out using the TEXT format (as opposed to CSV and BINARY).
+ * See the Postgres COPY documentation for more info.
+ * https://www.postgresql.org/docs/current/static/sql-copy.html
+ */
+class PostgresCopyWriter(private val wrapped: Writer) : Flushable by wrapped, Closeable by wrapped
+{
+    constructor(stream: OutputStream)
+        : this(stream.writer())
+
+    // From the documentation: "In particular, the following characters must be preceded by a backslash
+    // if they appear as part of a column value: backslash itself, newline, carriage return, and the current
+    // delimiter character."
+    private val illegalCharacters = Regex("\\|\n|\r|\t")
+
+    fun writeRecord(values: Iterable<Any?>)
+    {
```

Comment:
Can we have some unit tests for the escaping logic?

Reply:
Done, and they caught a bug, so thanks for catching this!

Reply:
Nice!